src/app/pages/SearchResultsPage/index.tsx

```diff
@@ -79,43 +77,41 @@ export const SearchResultsView: FC<{
         </SubPageCard>
       )}
 
-      {!isGlobal && !isAnyLoading && hasNoResultsOnWantedNetwork && <NoResults scope={wantedScope} />}
+      {!isGlobal && !isAnyLoading && hasNoResultsInWantedScope && <NoResults scope={wantedScope} />}
       {isGlobal && !isAnyLoading && hasNoResultsWhatsoever && <NoResults scope={wantedScope} />}
 
-      {!isGlobal && !isAnyLoading && !hasNoResultsOnWantedNetwork && (
-        <>
-          <ResultsOnNetwork
-            network={wantedScope.network}
-            searchQueries={searchQueries}
-            roseFiatValue={roseFiatValue}
-          />
-        </>
+      {!isGlobal && !isAnyLoading && !hasNoResultsInWantedScope && (
+        <ResultsFilteredThemed
+          title={getNameForScope(t, wantedScope)}
+          searchQueries={searchQueries}
+          networkForTheme={wantedScope.network}
+          numberOfResults={resultsInScopes[getKeyForScope(wantedScope)]}
+          roseFiatValue={roseFiatValue}
+          filter={Item => Item.network === wantedScope.network && Item.layer === wantedScope.layer}
```

Comment:
Related to https://github.com/oasisprotocol/explorer/pull/408#discussion_r1203133492

I would rather see the same logic for counting and filtering, e.g.
```tsx
export const SearchResultsPage {
  resultsInScopes: Record<string, HasScope[]>

  <ResultsFilteredThemed
    results={resultsInScopes[wantedScope]}

export const ResultsFilteredThemed {
  results.length
```

Reply:
Yes, that has been bothering me, too. As explained above, I have pending code elsewhere which will improve it. I think I'll just pull it here...

Reply:
We are no longer passing around the number of results, so this is solved.